### DISCUSSION TITLE
Standalone desktop application (with GUI) in Gleam

### DISCUSSION THREAD

**Discussion by sz3lbi** [gleam-lang/gleam#3249] [Questions & support]
Hi,  
I have the case where I need to create a completely standalone desktop application with a proper GUI. I thought this might be quite a nice opportunity to get more familiar with Gleam. What may be quite important is that I am not familiar with the Erlang environment at all.  
I couldn't find any GUI library (or bindings) available in Gleam. I can see that there is a module called `wx` in Erlang. However, in the language tour we can read about it:  
> External functions are useful but should be used sparingly. Prefer to write Gleam code where possible. 

Therefore, I assume that creating the GUI in a different technology can be a wise decision. Let's take Flutter as the option I'm looking forward to using.  
**How would you approach the data exchange between these two layers, considering that both would run locally as a desktop application?**  
Then I can bundle both and ship them together, I guess.  

Or should I stick with the `wx` in this case?

Importantly, I need this application to use native platform APIs, so any web-based solutions (similar to Electron) are out of the question.

**Comment by lpil** [comment]
wx and scenic are the most popular desktop GUI libraries for Erlang. You could also try ones made for JavaScript
